conn.go

```diff
@@ -955,14 +955,39 @@ func (c *Conn) WriteResponse(code int, enhCode EnhancedCode, text ...string) {
 		}
 	}
 
+	// Multiple tcp packets cause error
+	// in some controllers like sim800. reply must be a single tcp packet. 14.09.2020 ignatev vadim.
+	// example:
+	// s: 220 greets
+	// c: EHLO me
+	// s: 250-hello-me
+	// s: 250-SIZE 1111
+	// s: 250-AUTH LOGIN
+	// c: AUTH LOGIN      <- client got previous packet and began authorization
+	// s: 250-STARTTLS    <- server keeps sending reply to EHLO
+	// c: error, disconnect. beacuse expected 334 VXNlcm5hbWU6
+	// Essentially it is a client's problem. but...
```

Comment:
...but?

Reply:
...but  it's a simple fix. 
I totally agree with you. It's exactly why protocols were invented.
But keep in mind that those are widely used controllers. In a lot of serious industrial hardware. And by coincidence all(or most probably) of the public smtp servers like gmail are replying in a single packet. So they work fine. And this server is not working at all. It doesn't look very good from the view of the client, does it?
P.S. I wrote a claim to the manufacturer but chances on a new firmware or on a response at all are very slim.

Reply:
> And by coincidence all(or most probably) of the public smtp servers like gmail are replying in a single packet. 

I discovered that Go's net library disables Nagle's algorithm which is enabled by default in Linux. This is likely a reason most servers except for go-smtp send responses in a single packet - you can try doing `TCPConn.SetNoDelay(false)` on a corresponding client connection to see if starts working in the same way.